### PR TITLE
Debug admin verb: Pump Random Event

### PR DESCRIPTION
**Pull request by coiax**
Forces the event subsystem to make a new random event when it next
fires. (Most of the time this will be space dust.)

Used for testing the random event systems.

### CODE REVIEW

File: code/modules/admin/verbs/debug.dm [hunk 1]
```diff
@@ -771,4 +771,17 @@ var/list/TYPES_SHORTCUTS = list(
...
 		return
 
-	error_cache.show_to(src)
\ No newline at end of file
+	error_cache.show_to(src)
+
+/client/proc/pump_random_event()
+	set category = "Debug"
+	set name = "Pump Random Event"
+	set desc = "Scheudules the event subsystem to fire a new random event immediately."
```
Comment (lzimann) [inline code/modules/admin/verbs/debug.dm]: scheudules

**Comment by Cyberboss** [comment]
As a forewarning, a lot of events (i.e. space dust) give literally zero indication that they've run. You should explicitly state what event happened to the admins if this verb is used.

**Comment by coiax** [comment]
Given that events only run every three to ten minutes, maybe we could just take space dust and burnt out lights off the no-notify-admin list. What's one more line of information?

**Comment by Cyberboss** [comment]
They were removed originally for being too spammy

On Mar 17, 2017 2:06 PM, "coiax" <notifications@github.com> wrote:

Given that events only run every three to ten minutes, maybe we could just
take space dust and burnt out lights off the no-notify-admin list. What's
one more line of information?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/pull/25202#issuecomment-287429678>,
or mute the thread

**Comment by kevinz000** [comment]
Already possible via varediting however this is still good.

File: code/modules/admin/verbs/debug.dm (hunk 1 above)
Comment (octareenroon91) [inline code/modules/admin/verbs/debug.dm]: `schedules`

Comment (lzimann) [inline code/modules/admin/verbs/debug.dm]: seriously though fix the typo

Comment (octareenroon91) [inline code/modules/admin/verbs/debug.dm]: Also "schedule", "request", or "queue" might be a better verb than "pump".

**Pull request merged**